marlowe-playground-client/src/SimulationPage/View.purs

```diff
@@ -238,9 +238,20 @@ startSimulationWidget metadata { initialSlot, templateContent } =
                 [ text "Start simulation" ]
             ]
         ]
+  where
+  extractValueParameterNuberFormat valueParameter = case Map.lookup valueParameter metadata.valueParameterInfo of
+    Just { valueParameterFormat: DecimalFormat numDecimals currencyLabel } -> Just (currencyLabel /\ numDecimals)
+    _ -> Nothing
+
+  lookupDescription k m =
+    ( case Map.lookup k m of
+        Just { valueParameterDescription: description }
+          | trim description /= "" -> Just description
+        _ -> Nothing
+    )
 
-integerTemplateParameters :: forall action p. Map String String -> (IntegerTemplateType -> String -> BigInteger -> action) -> IntegerTemplateType -> String -> String -> Map String BigInteger -> Array (HTML p action)
-integerTemplateParameters explanations actionGen typeName title prefix content =
+integerTemplateParameters :: forall a action p. (String -> Maybe (String /\ Int)) -> (String -> Map String a -> Maybe String) -> Map String a -> (IntegerTemplateType -> String -> BigInteger -> action) -> IntegerTemplateType -> String -> String -> Map String BigInteger -> Array (HTML p action)
```

Comment:
same applies here. I think passing an object with params will make this easier to read